dev/Scroller/SnapPointWrapper.cpp

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "SnapPointWrapper.h"
+
+template<typename T>
```

Comment:
>template [](start = 0, length = 8)

I didn't see we need template here, why not just use winrt::ScrollSnapPointBase?

Reply:
SnapPointWrapper is used for both winrt::ScrollSnapPointBase and winrt::ZoomSnapPointBase, for homogeneous sets in Scroller.h:

std::set<std::shared_ptr<SnapPointWrapper<winrt::ScrollSnapPointBase>>, SnapPointWrapperComparator<winrt::ScrollSnapPointBase>> m_sortedConsolidatedHorizontalSnapPoints{};
std::set<std::shared_ptr<SnapPointWrapper<winrt::ScrollSnapPointBase>>, SnapPointWrapperComparator<winrt::ScrollSnapPointBase>> m_sortedConsolidatedVerticalSnapPoints{};
std::set<std::shared_ptr<SnapPointWrapper<winrt::ZoomSnapPointBase>>, SnapPointWrapperComparator<winrt::ZoomSnapPointBase>> m_sortedConsolidatedZoomSnapPoints{};

I'd rather keep using SnapPointWrapper<winrt::ScrollSnapPointBase> and SnapPointWrapper<winrt::ZoomSnapPointBase> rather than a non-templated SnapPointWrapper where SnapPoint returns the base winrt::SnapPointBase.